lldb/test/API/python_api/sbsavecoreoptions/TestSBSaveCoreOptions.py

```diff
@@ -26,3 +26,14 @@ def test_default_corestyle_behavior(self):
         """Test that the default core style is unspecified."""
         options = lldb.SBSaveCoreOptions()
         self.assertEqual(options.GetStyle(), lldb.eSaveCoreUnspecified)
+
+    def test_adding_and_removing_thread(self):
+        """Test adding and removing a thread from save core options."""
+        options = lldb.SBSaveCoreOptions()
+        options.AddThread(1)
```

Comment:
This won't work anymore right? We now require a SBThread. We need to do this test with a live process now. We also need to test a few more things:
- call `SBSaveCoreOptions::AddThread()` with a default constructed `SBThread` and verify error
- call `SBSaveCoreOptions::AddThread()` with a valid `SBThread` and then one from another process and verify the error. If you are loading a core file, you can use the live process and the core file process as your second process to verify these errors after you load the core file
- verify changing the process resets the thread specific options both with another valid SBProcess or an default constructed SBProcess.